Allow `alloc` feature to compile on stable with nom 6.0

Rustc v1.36.0 stabilized `extern crate alloc` and therefore the `alloc` feature gate is no longer necessary. Removing it would allow to compile `nom` with `alloc` feature and `no_std` mode on stable.

As nom currently defines rustc v1.31 as minimal supported version this is most likely a request for an upcoming nom 6.0.

----

- Rust version : rustc 1.38.0 (625451e37 2019-09-23)
- nom version : ^5.0
- nom compilation features used: {default-features = false, features = [ "alloc" ] }

```bash
> cargo build --no-default-features
   Compiling nom v5.0.1
error[E0554]: `#![feature]` may not be used on the stable release channel
   --> /home/rnd/.cargo/registry/src/github.com-1ecc6299db9ec823/nom-5.0.1/src/lib.rs:402:59
    |
402 | #![cfg_attr(all(not(feature = "std"), feature = "alloc"), feature(alloc))]
    |                                                           ^^^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0554`.
error: Could not compile `nom`.

To learn more, run the command again with --verbose.
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -8,7 +8,7 @@ rust:
   - nightly
   - beta
   - stable
-  - 1.31.0
+  - 1.36.0
 
 env:
   matrix:
@@ -20,12 +20,18 @@ before_script:
 
 matrix:
   include:
+    - rust: stable
+      env: FEATURES=''
+    - rust: stable
+      env: FEATURES='--no-default-features'
+    - rust: stable
+      env: FEATURES='--no-default-features --features "alloc"'
+    - rust: nightly
+      env: FEATURES=''
     - rust: nightly
       env: FEATURES='--no-default-features'
     - rust: nightly
       env: FEATURES='--no-default-features --features "alloc"'
-    - rust: stable
-      env: FEATURES=''
     - rust: nightly
       env: DOC_FEATURES='--features "std lexical regexp regexp_macros" --no-default-features'
       before_script:

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,11 @@
 
 ### Thanks
 
+### Changed
+
+- the minimal Rust version is now 1.36
+- stabilized the `alloc` feature
+
 ### Added
 
 ### Fixed

--- a/README.md
+++ b/README.md
@@ -5,7 +5,7 @@
 [![Build Status](https://travis-ci.org/Geal/nom.svg?branch=master)](https://travis-ci.org/Geal/nom)
 [![Coverage Status](https://coveralls.io/repos/Geal/nom/badge.svg?branch=master)](https://coveralls.io/r/Geal/nom?branch=master)
 [![Crates.io Version](https://img.shields.io/crates/v/nom.svg)](https://crates.io/crates/nom)
-[![Minimum rustc version](https://img.shields.io/badge/rustc-1.31.0+-lightgray.svg)](#rust-version-requirements)
+[![Minimum rustc version](https://img.shields.io/badge/rustc-1.36.0+-lightgray.svg)](#rust-version-requirements)
 
 nom is a parser combinators library written in Rust. Its goal is to provide tools
 to build safe parsers without compromising the speed or memory consumption. To
@@ -188,7 +188,7 @@ Some benchmarks are available on [Github](https://github.com/Geal/nom_benchmarks
 
 ## Rust version requirements
 
-The 5.0 series of nom requires **Rustc version 1.31 or greater**.
+The 5.0 series of nom requires **Rustc version 1.36 or greater**.
 
 Travis CI always has a build with a pinned version of Rustc matching the oldest supported Rust release.
 The current policy is that this will only be updated in the next major nom release.

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -399,7 +399,6 @@
 //! assert_eq!(alpha0_complete("abcd"), Ok(("", "abcd")));
 //! ```
 //! **Going further:** read the [guides](https://github.com/Geal/nom/tree/master/doc)!
-#![cfg_attr(all(not(feature = "std"), feature = "alloc"), feature(alloc))]
 #![cfg_attr(not(feature = "std"), no_std)]
 #![cfg_attr(feature = "cargo-clippy", allow(doc_markdown))]
 #![cfg_attr(nightly, feature(test))]

